File: app/src/main/java/era/apps/happinessjar/models/message/view_model/MessagesViewModel.java
package era.apps.happinessjar.models.message.view_model;


import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

import era.apps.happinessjar.models.message.data_base.AppMessage;

public class MessagesViewModel extends AndroidViewModel {
    private MessageRepository repository;
    private LiveData<List<AppMessage>> allAppMessage;
    private LiveData<List<AppMessage>> allAppWhatsApp;
    private LiveData<List<AppMessage>> allLikesMessage;


    public MessagesViewModel(@NonNull Application application) {
        super(application);
        repository = new MessageRepository(application);
        allAppMessage   = repository.getAllAppMessage();
        allAppWhatsApp  = repository.getAllAppWhatsApp();
        allLikesMessage = repository.getAllLikesMessage();
    }

    public LiveData<List<AppMessage>> getAllAppMessage() {
        return allAppMessage;
    }

    public LiveData<List<AppMessage>> getAllAppWhatsApp() {
        return allAppWhatsApp;
    }

    public LiveData<List<AppMessage>> getAllLikesMessage() {
        return allLikesMessage;
    }

    public void like(AppMessage message) {
        message.setStatus(message.getStatus()==1?0:1);
        repository.updateMessage(message);

    }


}

File: app/src/main/java/era/apps/happinessjar/ui/viewmodel/MessagesViewModel.kt
package era.apps.happinessjar.ui.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import era.apps.happinessjar.data.models.AppMessage
import era.apps.happinessjar.data.repository.MessageRepository
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch


class MessagesViewModel(application: Application) : AndroidViewModel(application) {

    private val repository: MessageRepository = MessageRepository(application)
    val allAppMessage: LiveData<List<AppMessage>> = repository.getAllAppMessage()
    val allAppWhatsApp: LiveData<List<AppMessage>> = repository.getAllAppWhatsApp()
    var allLikesMessage: LiveData<List<AppMessage>> = repository.getAllLikesMessage()


    fun like(message: AppMessage) = viewModelScope.launch {
        GlobalScope.launch {
            repository.updateMessage(message)
        }
//        allLikesMessage = repository.getAllLikesMessage()

    }

    fun insertToDataBase(message: AppMessage) = viewModelScope.launch {
        GlobalScope.launch {
            repository.insert(message)
        }
    }

    fun nullDelete() = viewModelScope.launch {
        GlobalScope.launch {
            repository.nullDelete()
        }
//        allLikesMessage = repository.getAllLikesMessage()

    }

}
